integrationtests/test_all_examples.py

```diff
@@ -0,0 +1,254 @@
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+
+""" Tests for demo and tutorial examples.
+"""
+
+import contextlib
+from itertools import chain
+import os
+import unittest
+from unittest import mock
+
+from traits.api import HasTraits
+
+from traitsui.tests._tools import (
+    is_current_backend_wx,
+    is_current_backend_qt4,
+    skip_if_null,
+)
+
+
+# This test file is not distributed nor is it in a package.
+HERE = os.path.dirname(__file__)
+
+EXAMPLES = os.path.join(HERE, "..", "examples")
+DEMO = os.path.join(EXAMPLES, "demo")
+TUTORIALS = os.path.join(EXAMPLES, "tutorials", "doc_examples", "examples")
+
+#: Explicitly include folders from which example files should be found
+#: recursively.
+SOURCE_DIRS = [
+    DEMO,
+    TUTORIALS,
+]
+
+
+#: Mapping from filepath to a callable() -> (skipped: bool, reason: str)
+FILES_MAY_BE_SKIPPED = {}
+
+
+def skip_file_if(condition, reason, filepath):
+    """ Mark a file to be skipped for a given condition.
+
+    Parameters
+    ----------
+    condition: callable() -> bool
+        The condition for skipping a file.
+    reason : str
+        Reason for skipping the file.
+    filepath : str
+        Path of the file which may be skipped from tests.
+    """
+    def wrapped():
+        return condition(), reason
+
+    filepath = os.path.abspath(filepath)
+    FILES_MAY_BE_SKIPPED[filepath] = wrapped
+
+
+skip_file_if(
+    lambda: True, "Not a target file to be tested",
+    os.path.join(DEMO, "demo.py"),
+)
+skip_file_if(
+    is_current_backend_wx, "ProgressRenderer is not implemented in wx.",
+    os.path.join(DEMO, "Advanced", "Table_editor_with_progress_column.py"),
+)
+skip_file_if(
+    lambda: not is_current_backend_wx(), "Only support wx",
+    os.path.join(DEMO, "Extras", "animated_GIF.py"),
+)
+skip_file_if(
+    lambda: not is_current_backend_qt4(), "Only support Qt",
+    os.path.join(DEMO, "Extras", "Tree_editor_with_TreeNodeRenderer.py"),
+)
+skip_file_if(
+    lambda: not is_current_backend_wx(), "Only support wx",
+    os.path.join(DEMO, "Extras", "windows", "flash.py"),
+)
+skip_file_if(
+    lambda: not is_current_backend_wx(), "Only support wx",
+    os.path.join(DEMO, "Extras", "windows", "internet_explorer.py"),
+)
+skip_file_if(
+    is_current_backend_wx,
+    "enable tries to import a missing constant. See enthought/enable#307",
+    os.path.join(DEMO, "Useful", "demo_group_size.py"),
+)
+skip_file_if(
+    lambda: True, "Require wx and is blocking.",
+    os.path.join(TUTORIALS, "view_multi_object.py"),
+)
+skip_file_if(
+    lambda: True, "Require wx and is blocking.",
+    os.path.join(TUTORIALS, "view_standalone.py"),
+)
```

Comment:
These two examples seem to me quite old. Should they be removed? 
Nevertheless I try to make this PR purely about adding tests.

Reply:
If there are no decisions for this, I will remove these two examples in a separate PR, partly because that is orthogonal and partly because then it can be reverted more easily should they need to be resurrected again.

Reply:
A curation needs to be done with the examples, which may involve removing some, but more likely just re-writing and modernizing.

The multi-object view is still valid, it just needs re-working an generalizing to use pyface GUI: the point is to show how to have multiple objects in the context of a view and to access their values).

Similarly the stand-alone example is showing that the view doesn't need to be part of the class.

Reply:
Thank you! I referred to this comment in #862 so that this can be taken into account.